src/main/java/edu/greenblitz/robotName/subsystems/Lifter/Lifter.java

```diff
@@ -0,0 +1,57 @@
+package edu.greenblitz.robotName.subsystems.Lifter;
+
+import com.revrobotics.CANSparkMax;
+import edu.greenblitz.robotName.utils.GBSubsystem;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class Lifter extends GBSubsystem {
+    private static Lifter instance;
+    private ILifter lifter;
+    private LifterInputs lifterInputs;
+    private ProfiledPIDController pid;
+
+    private Lifter() {
+        lifter = LifterFactory.create();
+        lifterInputs = new LifterInputs();
+        lifter.updateInputs(lifterInputs);
+        pid = new ProfiledPIDController(LifterConstants.PID_KP,LifterConstants.PID_KI,LifterConstants.PID_KD,new TrapezoidProfile.Constraints(LifterConstants.MAX_VELOCITY, LifterConstants.MAX_ACCELERATION));
+    }
+    public static Lifter getInstance() {
+        if(instance == null) {
+            instance = new Lifter();
+        }
+        return instance;
+    }
+    @Override
+    public void periodic() {
+        if(lifter.isSwitchPressed()) {
+            resetEncoderTo(0);
```

Comment:
why is this needed?

Reply:
כשהסוויטץ' נלחץ הוא אמור לאפס את האנקודר

Reply:
i prefer if you would create a function here that resets, another one that returns isPressed, and create a command that resets, and in the one that lowers the lifter put the isPressed as the is finish and reset in the end